File: frontend/src/types/recommendation.ts
export type RecommendationType = 'budget_pace' | 'category_trend' | 'savings_off_track'
export type RecommendationSeverity = 'info' | 'warning' | 'critical'

export interface Recommendation {
  type: RecommendationType
  severity: RecommendationSeverity
  message: string
  category_id: string | null
  goal_id: string | null
}

File: backend/app/schemas/recommendation.py
import uuid
from typing import Literal

from pydantic import BaseModel

RecommendationType = Literal["budget_pace", "category_trend", "savings_off_track"]
RecommendationSeverity = Literal["info", "warning", "critical"]


class Recommendation(BaseModel):
    type: RecommendationType
    severity: RecommendationSeverity
    message: str
    category_id: uuid.UUID | None = None
    goal_id: uuid.UUID | None = None


class RecommendationsResponse(BaseModel):
    recommendations: list[Recommendation]
